sources/api/early-boot-config/src/bin/early-boot-config.rs

```diff
@@ -91,6 +95,85 @@ fn parse_args(args: env::Args) -> Args {
     }
 }
 
+/// Gather user data providers to run in order
+fn gather_providers() -> Result<Vec<PathBuf>> {
+    Ok(WalkDir::new(PROVIDERS_DIR)
+        .max_depth(1)
+        .min_depth(1)
+        .sort_by_file_name()
+        .into_iter()
+        .collect::<std::result::Result<Vec<_>, _>>()?
```

Comment:
I'm a bit surprised that this error type doesn't need to be coaxed into a different one, but it's hard to tell in the diff view what the Error type of the returned Result is.

I would anticipate the iterator's error type to be `std::io::Error`.

Reply:
I was surprised as well, so I played around with this code a bit.

So what's going on here:
1. the iterator collects to a `Result<Vec<walkdir::DirEntry>, walkdir::Error>`
2. snafu implements `From<walkdir::Error>` on `error::Error` due to the `WalkDirSnafu` variant
3. the `?` operator automatically converts the `walkdir::Error` to `error::Error`

I guess I've been using `.context()` more than I need to, lol